Allow for longer student IDs and fix crash when students forget to logout

hdsbbrooks opened the pull request:
Our school uses 14 digit bar codes on student cards.  A pretty minor change but it might help someone else without the capacity to change and rebuild it themselves.

hdsbbrooks committed:
Allow for longer student IDs.  Our school uses 14 digit bar codes on student cards.

leeaidan commented:
Is there a way to develop a page in google sheets or an external GUI that displays a list of who is currently signed in? I feel as if this feature could benefit mentors or teachers who wanted to see who is in the lab at a certain time.

Daltz333 commented:
GrizzlyTime has a column that has shows which students are logged in or not in Google Sheets

![image](https://user-images.githubusercontent.com/10674555/68175855-95f77580-ff50-11e9-8ebe-60853b1c04b7.png)

hdsbbrooks committed:
Fix java crash bug if students forget to log out.

SEVERE: java.time.format.DateTimeParseException: Text '-6:12:38' could not be parsed at index 0

A negative value is checked for and the error flag is set but this code was running irrespective of the error flag status.  Moved the crashing lines so they only evaluate if the error flag is not set.

hdsbbrooks commented:
I ran into a crash bug when students forgot to log out for the day.  

java.time.format.DateTimeParseException: Text '-6:12:38' could not be parsed at index 0

A negative value is checked for and the error flag is set but this code was running irrespective of the error flag status.  Moved the crashing lines so they only evaluate if the error flag is not set.  I fixed it and added to this pull request.
